src/js/view/controls/rightclick.js

```diff
@@ -131,13 +132,21 @@ export default class RightClick {
             this.hideMenu();
             this.infoOverlay.open();
         };
+        if (this.shareOnRightClick) {
+            this.shareHandler = () => {
+                this.mouseOverContext = false;
+                this.hideMenu();
+                this.api.getPlugin('sharing').open();
+            };
+        }
     }
 
     setup(_model, _playerElement, layer) {
         this.playerElement = _playerElement;
         this.model = _model;
         this.mouseOverContext = false;
         this.layer = layer;
+        this.shareOnRightClick = (this.model.get('sharing') && this.model.get('sharing').shareOnRightClick);
```

Comment:
We don't really need a `shareOnRightClick` property. Just define `shareHandler` here, and use that and/or the presence of `'.jw-share-item'` to determine if sharing is enabled in this object.